src/main/java/com/maxmind/geoip2/model/EnterpriseResponse.java

```diff
@@ -14,19 +18,40 @@
  */
 public final class EnterpriseResponse extends AbstractCityResponse {
 
+    @MaxMindDbConstructor
     public EnterpriseResponse(
-            @JsonProperty("city") City city,
-            @JsonProperty("continent") Continent continent,
-            @JsonProperty("country") Country country,
-            @JsonProperty("location") Location location,
-            @JsonProperty("maxmind") MaxMind maxmind,
-            @JsonProperty("postal") Postal postal,
-            @JsonProperty("registered_country") Country registeredCountry,
-            @JsonProperty("represented_country") RepresentedCountry representedCountry,
-            @JsonProperty("subdivisions") ArrayList<Subdivision> subdivisions,
-            @JacksonInject("traits") @JsonProperty("traits") Traits traits
+            @JsonProperty("city") @MaxMindDbParameter(name="city") City city,
+            @JsonProperty("continent") @MaxMindDbParameter(name="continent") Continent continent,
+            @JsonProperty("country") @MaxMindDbParameter(name="country") Country country,
+            @JsonProperty("location") @MaxMindDbParameter(name="location") Location location,
+            @JsonProperty("maxmind") @MaxMindDbParameter(name="maxmind") MaxMind maxmind,
+            @JsonProperty("postal") @MaxMindDbParameter(name="postal") Postal postal,
+            @JsonProperty("registered_country") @MaxMindDbParameter(name="registered_country") Country registeredCountry,
+            @JsonProperty("represented_country") @MaxMindDbParameter(name="represented_country") RepresentedCountry representedCountry,
+            @JsonProperty("subdivisions") @MaxMindDbParameter(name="subdivisions") ArrayList<Subdivision> subdivisions,
+            @JacksonInject("traits") @JsonProperty("traits") @MaxMindDbParameter(name="traits") Traits traits
     ) {
         super(city, continent, country, location, maxmind, postal, registeredCountry,
                 representedCountry, subdivisions, traits);
     }
+
+    public EnterpriseResponse(
+            EnterpriseResponse response,
+            String ipAddress,
+            Network network,
+            List<String> locales
+    ) {
+        this(
+            response.getCity() != null ? new City(response.getCity(), locales) : null,
```

Comment:
We could get rid of some of the boilerplate by moving this stuff to the parent class.